cmd/nexusUpload.go

```
@@ -0,0 +1,256 @@
+	}
+	if err == nil {
+		artifactID := options.ArtifactID
+		if artifactID == "" {
+			artifactID = piperenv.GetParameter(".pipeline/commonPipelineEnvironment/configuration", "artifactId")
+		}
+		err = nexusClient.AddArtifact(nexus.ArtifactDescription{File: "mta.yaml", Type: "yaml", Classifier: "", ID: options.ArtifactID})
+	}
+	if err == nil {
+		mtarFilePath := piperenv.GetParameter(".pipeline/commonPipelineEnvironment", "mtarFilePath")
+		fmt.Println(mtarFilePath)
+		err = nexusClient.AddArtifact(nexus.ArtifactDescription{File: mtarFilePath, Type: "mtar", Classifier: "", ID: options.ArtifactID})
+	}
+	if err == nil {
+		err = nexusClient.UploadArtifacts()
+	}
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+type mtaYaml struct {
+	ID      string `json:"ID"`
+	Version string `json:"version"`
+}
+
+func setVersionFromMtaYaml(nexusClient *nexus.Upload) error {
+	var mtaYaml mtaYaml
+	mtaYamContent, err := ioutil.ReadFile("mta.yaml")
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(mtaYamContent, &mtaYaml)
+	if err != nil {
+		return err
+	}
+	return nexusClient.SetArtifactsVersion(mtaYaml.Version)
+}
+
+var errPomNotFound error = errors.New("pom.xml not found")
+
+func uploadMaven(nexusClient *nexus.Upload, options *nexusUploadOptions) {
+	err := uploadMavenArtifacts(nexusClient, options, "", "target", "")
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+
+	// Test if a sub-folder "application" exists and upload the artifacts from there as well.
+	// This means there are built-in assumptions about the project structure (archetype),
+	// that nexusUpload supports. To make this more flexible should be the scope of another PR.
+	err = uploadMavenArtifacts(nexusClient, options, "application", "application/target", options.AdditionalClassifiers)
+	if err == errPomNotFound {
+		// Ignore
+	} else if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
```

dito: let's bubble up the error ...